### PR TITLE
feat(test): add e2e tests for custom domain and gateway

### PR DESCRIPTION
**Pull request by TomaszSkrzyp**
**Description**

Changes proposed in this pull request:

- Add e2e tests for custom domain (Gardener-issued TLS, wildcard DNS, no-auth and OAuth2 APIRules) running on GCP and AWS Gardener clusters
- Add e2e tests for gateway: CR deletion blocking, Oathkeeper lifecycle, Kyma Gateway, VirtualService blocking and duplicate CR handling for Gardener and k3d
- Expand e2e helpers: resource assertions, LB target resolution, DNS readiness polling, and APIGateway CR lifecycle utilities
- Move non-secret config values from secrets to environmental variables in .github/workflows

**Pre-Merge Checklist**

Consider all the following items. If your contribution violates any of them, or you are not sure about it, add a comment to the PR.

- [x] The code coverage is acceptable.
- [x] Release notes for the introduced changes are created.
- [x] If Kubebuilder changes were made, you ran `make manifests` and committed the changes before the merge.
- [x] Pre-existing managed resources are correctly handled.
- [x] The change works on all hyperscalers supported by SAP BTP, Kyma runtime.
- [x] There is no upgrade downtime.
- [x] For infrastructure changes, you checked if the changes affect the hyperscaler's costs.
- [x] RBAC settings are as restrictive as possible.
- [x] If any new libraries are added, you verified license compliance and maintainability, and made a comment in the PR with details. We only allow stable releases to be included in the project.
- [x] You checked if this change should be cherry-picked to active release branches.
- [x] The configuration does not introduce any additional latency.
- [x] You checked if Busola updates are needed.

**Related issues**

### CODE REVIEW

File: tests/e2e/tests/custom_domain/custom_domain_test.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		assert.GreaterOrEqual(t, statusCode, http.StatusOK)
+		assert.Less(t, statusCode, http.StatusMultipleChoices)
+
+		statusCode, _, _, err = testBackground.Provider.MakeRequest(t, http.MethodGet, url, oauth2.WithTokenOverride("any-token"))
+		require.NoError(t, err, "Request with arbitrary token should succeed for noAuth endpoint")
+		assert.GreaterOrEqual(t, statusCode, http.StatusOK)
+		assert.Less(t, statusCode, http.StatusMultipleChoices)
+	})
+
+	// Implements following tests/integration scenarios:
```
Comment (kolodziejczak) [inline tests/e2e/tests/custom_domain/custom_domain_test.go]: I think we could avoid referencing this dir already

File: tests/e2e/tests/custom_domain/custom_domain_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	//   - Scenario: Calling a secured API with JWT and custom domain
```
Comment (kolodziejczak) [inline tests/e2e/tests/custom_domain/custom_domain_test.go]: This comment seems to be redundant

File: tests/e2e/tests/gateway/kyma_gateway_gardener/kyma_gateway_gardener_test.go
Comment (kolodziejczak) [inline tests/e2e/tests/gateway/kyma_gateway_gardener/kyma_gateway_gardener_test.go:1]: We don't have these tests running anywhere. We need to create a CICD job for that. Please keep them consistent with other Gardener based jobs if it comes to the target of the workflows.

File: tests/e2e/tests/gateway/deletion/deletion_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		modulehelpers.SetupBaseCR(t)
+
+		const oryRuleNamespace = "kyma-system"
+		const oryRuleName = "ory-rule"
+
+		_, err = infrahelpers.CreateResourceWithTemplateValues(
+			t,
+			ORYRule,
+			map[string]any{
+				"Name":      oryRuleName,
```
Comment (kolodziejczak) [inline tests/e2e/tests/gateway/deletion/deletion_test.go]: This name parameter is not present in the template

File: tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+		}
+		resourceasserts.AssertResourceExists(t, r, resourceasserts.StructCheck{Gvk: oathkeeperCRDResource.gvk, Name: oathkeeperCRDResource.name, Namespace: oathkeeperCRDResource.namespace}, checkTimeout)
+	})
+
+	t.Run("Kyma Gateway is not removed when there is a VirtualService", func(t *testing.T) {
+		modulehelpers.SetupBaseCR(t)
+		svcName, _, err := httpbinhelpers.DeployHttpbin(t, "kyma-system")
+		require.NoError(t, err, "Failed to deploy httpbin service")
+
+		vsName := "kyma-vs"
```
Comment (kolodziejczak) [inline tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go]: why this one is inconsistent with other arguments passed below? (extracted to var vs passing directly)

File: tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+		cr.Spec.EnableKymaGateway = new(false)
+		require.NoError(t, r.Update(context.Background(), &cr), "Failed to update APIGateway CR to disable Kyma Gateway")
+
+		istioGW := &unstructured.Unstructured{}
+		istioGW.SetGroupVersionKind(schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1beta1", Kind: "Gateway"})
+		istioGW.SetName("kyma-gateway")
+		istioGW.SetNamespace("kyma-system")
+		require.NoError(t, r.Get(context.Background(), "kyma-gateway", "kyma-system", istioGW))
+
+		require.NoError(t, modulehelpers.WaitUntilAPIGatewayExists(t, r, cr.GetNamespace(), cr.GetName()), "APIGateway CR should still exist while VirtualService is present")
```
Comment (kolodziejczak) [inline tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go]: why do we need to assert here that it exists?

File: tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+func TestKymaGateway(t *testing.T) {
+	t.Run("Oathkeeper is installed and uninstalled depending on APIGateway presence", func(t *testing.T) {
+		r, err := e2eclient.ResourcesClient(t)
+		require.NoError(t, err)
+		require.NoError(t, v1access.CreateAllowAPIRuleV1Signatures(context.Background(), r, t))
+		modulehelpers.SetupBaseCR(t)
+
+		kymaGatewayResource := resourceCheck{gvk: schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1beta1", Kind: "Gateway"}, name: "kyma-gateway", namespace: "kyma-system"}
+		resourceasserts.AssertResourceExists(t, r, resourceasserts.StructCheck{Gvk: kymaGatewayResource.gvk, Name: kymaGatewayResource.name, Namespace: kymaGatewayResource.namespace}, checkTimeout)
```
Comment (kolodziejczak) [inline tests/e2e/tests/gateway/kyma_gateway/kyma_gateway_test.go]: I think in the current state, we are losing check for the module annotation while asserting resources in the cluster. Probably in the other assertion places will be the same

**Review by kolodziejczak — DISMISSED** [review]

File: internal/reconciliations/oathkeeper/oathkeeper.go [hunk 7]
```diff
@@ -115,6 +115,7 @@ func DeleteOathkeeperIfNoRulesLeft(ctx context.Context, k8sClient client.Client)
 		deleteOathkeeperServices(ctx, k8sClient),
 		deleteDeployment(ctx, k8sClient, deploymentName),
 		deletePdb(ctx, k8sClient, pdbName, reconciliations.Namespace),
+		deleteCRD(ctx, k8sClient, crdName),
```
Comment (TomaszSkrzyp) [inline internal/reconciliations/oathkeeper/oathkeeper.go]: This is a bug fix introduced in other PR: https://github.com/kyma-project/api-gateway/pull/2811 
The bug was that the oathkeeper rules crd would not be deleted when no ORY rules were left

**Review by kolodziejczak — APPROVED** [review]
